circle.yml

```diff
@@ -0,0 +1,62 @@
+# Setup in CircleCI account the following ENV variables:
+# BINTRAY_ORGANIZATION
+# BINTRAY_ACCOUNT
+# BINTRAY_API_KEY
+general:
+  # Don't run CI for PR, only for major branches
+  branches:
+    only:
+      - master
+      - /v[0-9]+(\.[0-9]+)*/
+      - feature/circleci
+  artifacts:
+    - ~/packages
+
+machine:
+  environment:
+    ST2_PACKAGES_REPO: https://github.com/StackStorm/st2-packages
+  pre:
+    - mkdir -p ~/packages
+
+checkout:
+  post:
+    - git clone --depth 1 ${ST2_PACKAGES_REPO} /home/ubuntu/st2web/st2-packages
+
+dependencies:
+  pre:
+    - sudo apt-get install rpm
+    - npm install bower
+  post:
+    - node_modules/.bin/bower install
+
+test:
+  post:
+    - node_modules/.bin/gulp production
+
+deployment:
+  publish:
+    owner: StackStorm
+    branch:
+      - master
+      - /v[0-9]+(\.[0-9]+)*/
+      - feature/circleci
+    commands:
+      - dpkg-buildpackage -b -uc -us
+      - rpmbuild -bb rpm/st2web.spec
```

Comment:
Correct.

If we keep `build package` logic in deployment section there could be such scenario: 

- Open PR (`test` section passed -> `deployment` is not executed -> green)
- Merge PR into `master` (`test` passed -> `deployment` executed and failed -> red). Need to revert changes, fill another PR that will fix `rpmbuild` or `dpkg-buildpackage` logic.

By that reason it's better to move package creation logic into `tests` section and leave for `deployment` only simple uploading.